### ISSUE TITLE
How to make BART infill unmasked deletions (not masked tokens)?

### ISSUE THREAD

**Issue by d4buss** [huggingface/transformers#12539]
Hello,

I see that BART is able to infill <mask> tokens in the input and was able to run this example. https://huggingface.co/transformers/model_doc/bart.html#mask-filling

Since BART can also handle other type of pre-training noise such as token deletion, is there a way I could get BART to take inputs with deletions, and see the output of BARTs attempt to infill those deleted spans?

In the BART paper, 
`A _C. _E.   ->. A B C. D E. `    is an example of denoising Token Masking, and 
`A. C.  E.     ->  A B C. D E. `     is an example of denoising Token Deletion 

Since the best BART model in the paper is found to be pre-trained with "sentence shuffling" and "text-infilling,
where arbitrary length spans of text are replaced with a single mask token," and NOT with the token deletion strategy (where no <mask> token is left behind), would we expect to have to obtain a variant of BART that has been pre-trained on token deletion in order to use BART for infilling unmasked deletions?

I am asking because I would like to use BART to repair text where certain spans may be missing/ deleted. For example:
input: `"Wikipedia is a free online, created and edited by volunteers around the world"`
desired output:  `"Wikipedia is a free online encyclopedia, created and edited by volunteers around the world"`
(Note the generation of "encyclopedia" but no <mask> token included in the input.)
